zenodo/modules/metrics/utils.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2017-2021 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Utilities for metrics module."""
+from flask import Response
+from . import config
+
+
+def openaire_response():
+    response = ''
+    for metric in config.ZENODO_METRICS['openaire-nexus']:
+        response += "# HELP {name} {help}\n# TYPE {name} {type}\n{name} " \
+                    "{value}\n".format(**metric)
+    return Response(response, mimetype='text/plain')
+
+
+def formatted_response(metric_id):
+    formatters = {
+        'openaire-nexus': openaire_response
+    }
+
+    response = formatters.get(metric_id)
+
+    return Response('Invalid key', status=400, mimetype='text/plain') if \
```

Comment:
HTTP formatting can be done in the `views.py`. This function can e.g. just return the formatted text result.